src/OpenIddict.EntityFramework/Stores/OpenIddictApplicationStore.cs

```diff
@@ -60,19 +60,19 @@ public class OpenIddictApplicationStore<TApplication, TToken, TContext, TKey> :
 
             await Context.SaveChangesAsync(cancellationToken);
 
-            return converter.ConvertToInvariantString(application.Id);
+            return converter.ConvertToInvariantString(application.ApplicationId);
         }
 
         /// <summary>
-        /// Retrieves an application using its unique identifier.
+        /// Finds and returns an application, if any, which has the specified <paramref name="id"/>.
         /// </summary>
-        /// <param name="identifier">The unique identifier associated with the application.</param>
+        /// <param name="id">The ID of application to search for.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> that can be used to abort the operation.</param>
         /// <returns>
         /// A <see cref="Task"/> that can be used to monitor the asynchronous operation,
         /// whose result returns the client application corresponding to the identifier.
         /// </returns>
-        public virtual Task<TApplication> FindByIdAsync(string identifier, CancellationToken cancellationToken) {
+        public virtual Task<TApplication> FindByIdAsync(string id, CancellationToken cancellationToken) {
```

Comment:
Added one more method for that purpose.
Kept this one for style compatibility with Identity reason.